backend/stylists/app/Http/Controllers/RecommendationController.php

```diff
@@ -0,0 +1,126 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use Illuminate\Http\Request;
+
+use App\Http\Requests;
+use Illuminate\Support\Facades\Auth;
+use Validator;
+use App\Recommendation;
+use App\Push;
+use App\Client;
+use App\Stylist;
+use App\Look;
+use App\Product;
+use App\Models\Lookups\EntityType;
+use App\Models\Enums\AppSections;
+use App\Models\Enums\RecommendationType;
+use App\Models\Enums\EntityType as EntityTypeId;
+
+class RecommendationController extends Controller
+{
+    public function index(Request $request, $action, $id = null, $action_id = null)
+    {
+        $method = strtolower($request->method()) . strtoupper(substr($action, 0, 1)) . substr($action, 1);
+        if ($id) {
+            $this->resource_id = $id;
+        }
+        if ($action_id) {
+            $this->action_resource_id = $action_id;
+        }
+        return $this->$method($request);
+    }
+
+    public function postSend(Request $request)
+    {
+        $app_section = isset($request->app_section) && $request->app_section != "" ? $request->app_section : AppSections::MY_REQUESTS;
+
+        $errormsg = '';
+        $entity_type_id = '';
+        if(isset($request->entity_type_id)){
+            $entity_type_id = $request->entity_type_id;
+        }else{
+            $errormsg = 'Entity type undefined';
+        }
+
+        $client_ids = isset($request->client_ids) ? $request->client_ids : '';
+        $entity_ids = isset($request->entity_ids) ? $request->entity_ids : '';
+
+        $stylish_id = Auth::user()->stylish_id;
+        $entity_type_name = EntityType::where('id', $entity_type_id)->first()->name;
+        if($entity_type_id) {
+            foreach ($client_ids as $client) {
+                $message_pushed = 0;
+                foreach ($entity_ids as $entity_id) {
+                    $recommends_arr[] = array(
+                        'user_id' => $client,
+                        'recommendation_type_id' => RecommendationType::MANUAL,
+                        'created_by' => $stylish_id,
+                        'entity_type_id' => $entity_type_id,
+                        'entity_id' => $entity_id
+                    );
+
+                    if ($message_pushed == 0) {
+                        $client_data = Client::where('user_id', $client)->first();
+                        $stylist_data = Stylist::where('stylish_id', $stylish_id)->first();
+
+                        if (empty($client_data) || empty($stylist_data)) {
+                            break;
+                        }
+                        $regId = $client_data->regId;
+                        $stylish_name = $stylist_data->name;
+                        $entity_url = '';
+                        if ($entity_type_id == EntityTypeId::PRODUCT) {
```

Comment:
Fetch all entities in one query and store the entities in an array. Use this array as lookup.

No db queries inside a loop.